Examples/ArcGISToolkitExamples/TemplatePickerExample.swift

```diff
@@ -0,0 +1,109 @@
+        map?.operationalLayers.add(featureLayer)
+
+        // Here we give the feature layer a default popup definition.
+        // We have to load it first to create a default popup definition.
+        // If you create the map from a portal item, you can define the popup definition
+        // in the webmap and avoid this step.
+        featureLayer.load{ _ in
+            featureLayer.popupDefinition = AGSPopupDefinition(popupSource: featureLayer)
+        }
+        
+        // set the map on the mapview
+        mapView.map = map
+        
+        // Log if there is any error loading the map
+        map?.load{ error in
+            if let error = error{
+                print("error loading map: \(error)")
+            }
+        }
+        
+        // add bar button item for showing templates
+        let bbi = UIBarButtonItem(title: "Templates", style: .plain, target: self, action: #selector(showTemplates))
+        navigationItem.rightBarButtonItem = bbi
+    }
+    
+    @objc private func showTemplates(){
+        
+        guard let map = map else { return }
+        
+        // Instantiate the TemplatePickerViewController
+        let templatePicker = TemplatePickerViewController(map: map)
+        
+        // Assign the delegate
+        templatePicker.delegate = self
+        
+        // Present the template picker
+        self.navigationController?.pushViewController(templatePicker, animated: true)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    
+}
+
+extension TemplatePickerExample: TemplatePickerViewControllerDelegate {
+    
+    public func templatePickerViewControllerDidCancel(_ templatePickerViewController: TemplatePickerViewController) {
+        
+        // This is where you handle the user canceling the template picker
+        
+        // dismiss the template picker
+        navigationController?.popViewController(animated: true)
+        
+        let alert = UIAlertController(title: "TemplatePickerExample", message: "User cancelled", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
+        
+        present(alert, animated: true, completion: nil)
+        
+    }
+    
+    public func templatePickerViewControllerDidSelectTemplate(_ templatePickerViewController: TemplatePickerViewController, featureTemplateInfo: FeatureTemplateInfo) {
+        
+        // This is where you handle the user making a selection with the template picker
+        
+        // dismiss the template picker
+        navigationController?.popViewController(animated: true)
+        
+        let alert = UIAlertController(title: "TemplatePickerExample", message: "User selected \(featureTemplateInfo.featureTemplate.name)", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
```

Comment:
how do I do that?

Reply:
Save the action to a temporary variable and then set it as the preferred action after adding it to the alert controller:

```swift
let okayAction = UIAlertAction(title: "Ok", style: .default)
let alert = UIAlertController(title: "TemplatePickerExample", message: "User selected \(featureTemplateInfo.featureTemplate.name)", preferredStyle: .alert)
alert.addAction(okayAction)
alert.preferredAction = okayAction
```

FYI: I've noticed that Apple uses "OK" rather than "Ok" for the title of buttons like this.